### ISSUE TITLE
Windows install showing exception when excluding files in the build process.

### ISSUE THREAD

**Issue by cupples** [tintinweb/scapy-ssl_tls#88]
The current setup file from Latest commit 03c1413  on Nov 17
Python 2.7.9 windows 10
```python
C:\Python27\python.exe C:/Users/Shawn/Downloads/scapy-ssl_tls-master/scapy-ssl_tls-master/setup.py install
INFO: Installing scapy-ssl_tls layers to: []
Traceback (most recent call last):
  File "C:/Users/Shawn/Downloads/scapy-ssl_tls-master/scapy-ssl_tls-master/setup.py", line 158, in <module>
    data_files=get_layer_files_dst(get_site_packages()),
  File "C:/Users/Shawn/Downloads/scapy-ssl_tls-master/scapy-ssl_tls-master/setup.py", line 66, in get_layer_files_dst
    for layer_file in os.listdir(path):
WindowsError: [Error 3] The system cannot find the path specified: 'scapy_ssl_tls/*.*'
Process finished with exit code 1
```

making the following change to the function solves the exception though I am still hitting an error (turned out to be a env pathing issue):
```python
def get_layer_files_dst(sites, path="scapy_ssl_tls"):
    data_files = []
    scapy_locations = get_scapy_locations(sites)
    layer_files = []
    cwd = os.path.dirname(os.path.realpath(__file__))
    for layer_file in os.listdir(os.path.join(cwd,path)):
        # Copy only python files, and exclude module file from copy to scapy
        if layer_file != "__init__.py" and layer_file.endswith(".py"):
            layer_file_path = os.path.join(path, layer_file)
            if os.path.isfile(layer_file_path):
                layer_files.append(layer_file_path)
    for scapy_location in scapy_locations:
        data_files.append(
            (os.path.join(scapy_location, "layers"), layer_files))
    return data_files
```

**Comment by alexmgr** [comment]
Hi @cupples,

Thanks for taking the time to report this. I edited your post to add syntax highlighting.
Yes indeed, it looks like we only considered the case, where `setup.py` would be called from the `scapy-ssl_tls` folder. Once #85 is merged, I'll make sure to push your change in.

I'm not clear if this solved your issue overall though. Are you all set once `setup.py` is patched?

**Comment by cupples** [comment]
Thanks Alex,
I am all set. I did wind up needing to install and deal with a bunch of other packages to get it going but thats my own fault for using Windows for python. As I needed ultimately to install everything I would consider this a low pri issue.

**Comment by tintinweb** [comment]
Hi,

_//not_ specific to windows.

I am not quite sure if `setup.py` is intended to be run outside the project directory as path information in `setup.py` is usually configured relative to the project directory. 

In order to address this, we would have to fix `get_layer_files_dst()` as well as `setup(..., packages=['scapy_ssl_tls'], ...)` `packages`-param otherwise `setup.py` would fail with the following error:

    error: package directory 'scapy_ssl_tls' does not exist

But fixing the `packages` param would raise the following warning:

    WARNING: '/home/tin/scapy-ssl_tls/scapy_ssl_tls' not a valid package name; please use only .-separated package names in setup.py
    running install
    ...

`setup.py` is obviously not happy about that misconfiguration.
Any thoughts?

_For reference, this is my patch:_
```diff
diff --git a/setup.py b/setup.py
index d4b5f01..a7d8a61 100755
--- a/setup.py
+++ b/setup.py
@@ -12,6 +12,11 @@ from setuptools.command.install import install as _install
 import site as _site
 import pkg_resources

+def fix_path(path):
+    if os.path.exists(path):
+        return path
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)), path)
+
 def get_site_packages():
     """
     This is a hack to work around site.getsitepackages() not working in
@@ -63,7 +68,7 @@ def get_layer_files_dst(sites, path="scapy_ssl_tls"):
     data_files = []
     scapy_locations = get_scapy_locations(sites)
     layer_files = []
-    for layer_file in os.listdir(path):
+    for layer_file in os.listdir(fix_path(path)):
         # Copy only python files, and exclude module file from copy to scapy
         if layer_file != "__init__.py" and layer_file.endswith(".py"):
             layer_file_path = os.path.join(path, layer_file)
@@ -139,7 +144,7 @@ except (pkg_resources.DistributionNotFound, pkg_resources.VersionConflict):
 setup(
     name="scapy-ssl_tls",
     version="1.3.0",
-    packages=["scapy_ssl_tls"],
+    packages=[fix_path("scapy_ssl_tls")],
     author="tintinweb",
```

**Comment by alexmgr** [comment]
Agree with you @tintinweb. Maybe we should just leave as is and close this off?

**Comment by tintinweb** [comment]
Thanks for the report @cupples. I am going to close this as invalid. Please make sure to always run `setup.py` from within the project dir otherwise lots of crazy things may happen :) Feel free to open another issue if you have problems setting things up on windows.
